File: sentinelx-frontend/types/signal-detail.ts
import type { IntelligenceSignal } from '@/types/sentinelx';

export type SourceProvenance = {
  source_id: string;
  source_name: string;
  source_url: string;
  source_type: string;
  category?: string | null;
};

export type RecordExcerpt = {
  parsed_record_id: string;
  title?: string | null;
  excerpt: string;
  url?: string | null;
  fetched_at?: string | null;
  content_hash?: string | null;
};

export type StoryEvent = {
  id: string;
  event_type: string;
  title: string;
  summary: string;
  occurred_at: string;
  severity: number;
  source_label?: string | null;
};

export type HumanNote = {
  id: string;
  team_role: string;
  author_name: string;
  content: string;
  created_at: string;
};

export type SignalDetail = {
  signal: IntelligenceSignal;
  source: SourceProvenance;
  record: RecordExcerpt;
  related_signals: IntelligenceSignal[];
  story_timeline: StoryEvent[];
  human_notes: HumanNote[];
};

export type WorkspaceProfile = {
  id: string;
  slug: string;
  company_name: string;
  profile: Record<string, unknown>;
  updated_at: string;
};

File: sentinelx-backend/app/schemas/signal_detail.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.signal import IntelligenceSignalRead


class SourceProvenance(BaseModel):
    source_id: UUID
    source_name: str
    source_url: str
    source_type: str
    category: str | None = None


class RecordExcerpt(BaseModel):
    parsed_record_id: UUID
    title: str | None = None
    excerpt: str = ""
    url: str | None = None
    fetched_at: datetime | None = None
    content_hash: str | None = None


class StoryEvent(BaseModel):
    id: str
    event_type: str
    title: str
    summary: str
    occurred_at: datetime
    severity: int = 0
    source_label: str | None = None


class SignalDetailResponse(BaseModel):
    signal: IntelligenceSignalRead
    source: SourceProvenance
    record: RecordExcerpt
    related_signals: list[IntelligenceSignalRead] = Field(default_factory=list)
    story_timeline: list[StoryEvent] = Field(default_factory=list)
    human_notes: list[dict] = Field(default_factory=list)
